pkg/api/store/projectsetter/store.go

```diff
@@ -43,6 +64,63 @@ func (t *transformer) list(apiContext *types.APIContext, schema *types.Schema, d
 	return data, nil
 }
 
+// Retrieves project from api sub-context, then finds associated namespaces. If found, assigns to options.
+func (p projectSetter) setOptionsNamespaces(apiContext *types.APIContext, opt *types.QueryOptions) error {
+	// should only query by namespaces for get requests
+	if apiContext.Method != http.MethodGet {
+		return nil
+	}
+
+	clusterName := p.ClusterManager.ClusterName(apiContext)
+	if clusterName == "" {
+		return nil
+	}
+
+	clusterContext, err := p.ClusterManager.UserContext(clusterName)
+	if err != nil {
+		return err
+	}
+
+	namespaces, err := clusterContext.Core.Namespaces("").Controller().Lister().List("", labels.NewSelector())
+	if err != nil {
+		return err
+	}
+
+	matchingNamespaces, err := getMatchingNamespaces(*apiContext, namespaces)
+
+	if opt == nil {
+		*opt = types.QueryOptions{}
```

Comment:
common practice is to write this way `opt = &types.QueryOptions{}`, but not a big deal.